dbt_platform_helper/templates/pipelines/codebase/overrides/stack.ts

```diff
@@ -122,6 +122,9 @@ export class TransformedStack extends cdk.Stack {
                 )),
             },
         });
+
+        imageBuildProject.node.addDependency(this.template.getResource("BuildProjectRole") as cdk.aws_iam.CfnRole);
+        imageBuildProject.node.addDependency(this.template.getResource("BuildProjectPolicy") as cdk.aws_iam.CfnPolicy);
```

Comment:
There was also a bit of a race condition when the pipeline is first created where it sometimes tried to run checks before the role/policy were fully there. These dependencies resolve that.